google-cloud-bigtable/lib/google/cloud/bigtable/status.rb

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Bigtable
+      ##
+      # # Status
+      #
+      # Represents a logical error model from the Bigtable service, containing an
+      # error code, an error message, and optional error details.
+      #
+      # @attr [Integer] code The status code, which should be an enum value of
+      #   [google.rpc.Code](https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto).
+      # @attr [String] description The human-readable description for the status
+      #   code, which should be an enum value of
+      #   [google.rpc.Code](https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto).
+      #   For example, `INVALID_ARGUMENT`.
+      # @attr [String] message A developer-facing error message, which should be
+      #   in English.
+      # @attr [Array, nil] details A list of messages that carry the error
+      #   details.
+      #
+      # @example
+      #   require "google/cloud/bigtable"
+      #
+      #   bigtable = Google::Cloud::Bigtable.new
+      #
+      #   table = bigtable.table("my-instance", "my-table")
+      #
+      #   entries = []
+      #   entries << table.new_mutation_entry("row-1").set_cell("cf1", "field1", "XYZ")
+      #   entries << table.new_mutation_entry("row-2").set_cell("cf1", "field1", "ABC")
+      #   responses = table.mutate_rows(entries)
+      #
+      #   responses.each do |response|
+      #     puts response.status
```

Comment:
If we're encouraging people to print status objects as strings, should it have a formatted `to_s`? Otherwise, it might be better to print just one of the fields.

Reply:
Good call, will print just one field.